### ISSUE TITLE
Memory leak?

### ISSUE THREAD

**Issue by busybox11** [docterd/MusaicFM#50]
Hi!

I've been using this project for a couple of months now. I've been experiencing lots of memory usage by the screen saver in the background even when it's not shown - around 1 to 1.5GB. Sometimes I have to kill the program to free up some memory. 

I don't know how to debug this, as I don't have many experience with Apple APIs and software overall. I would love to help though!

**Comment by obrhoff** [comment]
Is the Legacy Screen Saver? I've seen it myself, eating up the memory. But I think it can't be this screen saver since the process itself is stopped, after leaving the screensaver.

**Comment by busybox11** [comment]
Yes, legacy screen saver. Finally caught it after forgetting to start Amphetamine today.

<img width="1046" alt="image" src="https://github.com/obrhoff/MusaicFM/assets/29630035/cc5b7df6-2629-4610-98df-15069aa02e79">

What could cause this?

**Comment by obrhoff** [comment]
Idk. The ScreenSaver itself is already pretty old, but I noticed this issue coming up recently (maybe in the last 2 years).

Maybe its a bug from Apple side. Hard to tell.

**Comment by busybox11** [comment]
No idea. I've been using a Macbook for less than a year, and have always had this

Will keep an eye on it then. Thanks for your feedback!

**Comment by obrhoff** [comment]
Its definitely a problem with the LegacyScreenSaver. 

https://github.com/vpeschenkov/Predator/issues/13

**Comment by yazidears** [comment]
Hey! I managed to track down the memory leak in Predator, and they’ve just merged my pull request, so it should be fixed now! 🚀

**Comment by obrhoff** [comment]
Thanks for pointing that out. I just merged this PR and published a new version. Can you test?

https://github.com/obrhoff/MusaicFM/pull/56
